Orange/classification/_tree_scorers.pyx

```diff
@@ -0,0 +1,293 @@
+#cython: embedsignature=True
+
+import numpy
+cimport numpy as np
+import cython
+
+from libc.stdlib cimport calloc, free
+
+cdef extern from "math.h":
+    double log(double x)
+
+cdef extern from "numpy/npy_math.h":
+    bint npy_isnan(double x)
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def find_threshold_entropy(np.ndarray[np.float64_t, ndim=1] x,
+                           np.ndarray[np.float64_t, ndim=1] y,
+                           np.ndarray[np.int64_t, ndim=1] idx,
+                           int n_classes, int min_leaf):
+    cdef unsigned int *distr = <unsigned int *>calloc(2 * n_classes, sizeof(int))
+    cdef Py_ssize_t i, j
+    cdef double entro, class_entro, best_entro
+    cdef unsigned int p, curr_y
+    cdef unsigned int best_idx = 0
+    cdef unsigned int N = idx.shape[0]
```

Comment:
You can wrap all cdefs in a single `cdef` block: http://docs.cython.org/en/latest/src/userguide/language_basics.html#grouping-multiple-c-declarations